### PR TITLE
iconnconfig: init at 0.5.2-beta

### PR DESCRIPTION
**Pull request by ezemtsov**
##### Motivation for this change

iConnConfig - an open source version of control software for iConnectivity devices.
Tested with my iConnectMidi4+

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by hugolgst — CHANGES_REQUESTED** [review]
Seems to be a problem with fetching your Git repository:
```
Unable to checkout refs/tags/0.5.2-beta from https://codeberg.org/dehnhardt/IConnConfig.
```

Result of `nixpkgs-review pr 99608`:

```
1 package failed to build:
iconnconfig
```

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, fetchgit
+, qt512
+, alsaLib
+, libusb1
+}:
+
+with qt512;
```
Comment (worldofpeace) [inline pkgs/applications/audio/iconnconfig/default.nix:8]: please use the `libsForQt512.callPackage` function. And is there any specific reason why it needs to be that version? Also note, by using that callPackage function the attributes like `qtbase` are going to be in the lexical scope. So you will have to replace `qt512` with all the attributes in the scope of qt512 in the arguments list above.

Comment (ezemtsov) [reply pkgs/applications/audio/iconnconfig/default.nix]: Replaced with libsForQt5 in all-packages

**Comment by ezemtsov** [comment]
> Seems to be a problem with fetching your Git repository:
> 
> ```
> Unable to checkout refs/tags/0.5.2-beta from https://codeberg.org/dehnhardt/IConnConfig.
> ```
> 
> Result of `nixpkgs-review pr 99608`:
> 
> ```
> 1 package failed to build:
> iconnconfig
> ```

Added missing "v" prefix to fetchGit

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  unpackPhase = ''
+    mkdir -p source/iconnconfig
+    mkdir -p source/PKControls
+    cp -r $src/* source/iconnconfig
+    cp -r $PKControls/* source/PKControls
+  '';
+
+  sourceRoot = "source/iconnconfig";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:31]: Can this be exported in the pre section where it is used?

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+mkDerivation rec {
+  pname = "IConnConfig";
+  version = "0.5.2-beta";
+  src = fetchgit {
+    url = "https://codeberg.org/dehnhardt/IConnConfig";
+    rev = "v${version}";
+    sha256 = "10g0cgqkpqgwnbjdndhmyk4h680kbmlws3mc678fmgjy75qvbfmj";
+  };
+
+  PKControls = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:18]: is this useful outside this package? How is it going to be updated?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  qmakeFlags = [ "iConnConfig.pro" "CONFIG+=build_deb" ];
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://codeberg.org/dehnhardt/IConnConfig";
+    description = "Linux based configuration utility for IConnectivity interfaces";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ezemtsov ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:44]: Please add yourself to the maintaienrs list and commit it in a separate commit with the message ``maintainers: add ezemtsov``.

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    url = "https://codeberg.org/dehnhardt/PKControls.git";
+    rev = "f49a0105b65c19e3f63962a1e242839be15bc6b6";
+    sha256 = "1n6lkqy9ihqffs1s2ihdzzhk7dxi1cjj4igs3m00m8h0b5g78gga";
+  };
+
+  sourceRoot = "source/iconnconfig";
+
+  unpackPhase = ''
+    mkdir -p ${sourceRoot}
+    mkdir -p source/PKControls
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:28]:
```suggestion
    mkdir -p ${sourceRoot} source/PKControls
```

**Comment by SuperSandro2000** [comment]
Please also comment on https://github.com/NixOS/nixpkgs/pull/99598#discussion_r531806295

**Comment by SuperSandro2000** [comment]
Please squash the review commit into the package.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99598` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iconnconfig</li>
  </ul>
</details>

**Comment by ezemtsov** [comment]
> Please squash the review commit into the package.

done

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 99598` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>iconnconfig</li>
  </ul>
</details>

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+mkDerivation rec {
+  pname = "IConnConfig";
+  version = "0.5.2-beta";
+  src = fetchgit {
+    url = "https://codeberg.org/dehnhardt/IConnConfig";
+    rev = "v${version}";
+    sha256 = "10g0cgqkpqgwnbjdndhmyk4h680kbmlws3mc678fmgjy75qvbfmj";
+  };
+
+  PKControls = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:18]: Those should use fetchFromGitea with the domain pointing to codeberg.

File: pkgs/applications/audio/iconnconfig/default.nix [hunk 7]
```diff
@@ -0,0 +1,46 @@
+{ lib
+, mkDerivation
+, fetchgit
+, qmake
+, alsaLib
+, libusb1
+}:
+
+mkDerivation rec {
+  pname = "IConnConfig";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/iconnconfig/default.nix:10]:
```suggestion
  pname = "iconnconfig";
```

the attr and pname should ideally match.

**Comment by SuperSandro2000** [comment]
Closing as author usually takes very long to respond to review feedback and this open since years.

**Pull request closed without merging**